plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,304 @@
+			sprintf(
+				/* translators: %s is the error code */
+				__( 'The REST API responded with the error code <code>%s</code> and the following error message:', 'optimization-detective' ),
+				esc_html( (string) $response->get_error_code() )
+			),
+			array( 'code' => array() )
+		) . '</p><blockquote>' . esc_html( $response->get_error_message() ) . '</blockquote>';
+	} else {
+		$code    = wp_remote_retrieve_response_code( $response );
+		$message = wp_remote_retrieve_response_message( $response );
+		$body    = wp_remote_retrieve_body( $response );
+		$data    = json_decode( $body, true );
+
+		$required_args = array( 'slug', 'current_etag', 'hmac', 'url', 'viewport', 'elements' );
+		$is_expected   = (
+			400 === $code &&
+			isset( $data['code'], $data['data']['params'] ) &&
+			'rest_missing_callback_param' === $data['code'] &&
+			is_array( $data['data']['params'] ) &&
+			count( array_intersect( $data['data']['params'], $required_args ) ) === count( $required_args )
+		);
+		if ( ! $is_expected ) {
+			$result['status']      = 'critical';
+			$result['label']       = __( 'The Optimization Detective REST API endpoint is unavailable to logged-out users', 'optimization-detective' );
+			$result['description'] = $common_description_html . $error_description_html . '<p>' . wp_kses(
+				sprintf(
+					/* translators: %d is the HTTP status code, %s is the status header description */
+					__( 'The REST API returned with an HTTP status of <code>%1$d %2$s</code>.', 'optimization-detective' ),
+					$code,
+					esc_html( $message )
+				),
+				array( 'code' => array() )
+			) . '</p>';
+
+			if ( isset( $data['message'] ) && is_string( $data['message'] ) ) {
+				$result['description'] .= '<blockquote>' . esc_html( $data['message'] ) . '</blockquote>';
+			}
+
+			$result['description'] .= '<details><summary>' . esc_html__( 'Raw response:', 'optimization-detective' ) . '</summary><pre style="white-space: pre-wrap">' . esc_html( $body ) . '</pre></details>';
+		}
+	}
+	return $result;
+}
+
+/**
+ * Gets the response to an Optimization Detective REST API store request to confirm it is available to unauthenticated requests.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @param bool $use_cached Whether to use a previous response cached in a transient.
+ * @return array{ response: array{ code: int, message: string }, body: string }|WP_Error Response.
+ */
+function od_get_rest_api_health_check_response( bool $use_cached ) {
+	$transient_key = 'od_rest_api_health_check_response';
+	$response      = $use_cached ? get_transient( $transient_key ) : null;
+	if (
+		(
+			is_array( $response )
+			&&
+			isset( $response['response']['code'], $response['response']['message'], $response['body'] )
+			&&
+			is_int( $response['response']['code'] )
+			&&
+			is_string( $response['response']['message'] )
+			&&
+			is_string( $response['body'] )
+		)
+		||
+		is_wp_error( $response )
+	) {
```

Comment:
Why do we check all this? Seems overkill, a valid transient result should simply be not `false`. Also I'm not sure how the `is_wp_error()` clause would ever return `true`? A cached value cannot be a `WP_Error` object I'm pretty sure?

Why not make this simpler?
```suggestion
	$response      = $use_cached ? get_transient( $transient_key ) : false;
	if ( false !== $response ) {
```

Per my other comment about caching a `WP_Error`, we then should also add logic to turn the raw data from an error back into an actual `WP_Error` object.

Reply:
Yes, maybe this type checking is overkill. Removed in b4736a40553deb6114ed96c28d94759cb38b1076. I added these safeguards since the transient can technically return anything.

A cached value can be a `WP_Error`, since the values stored in transients are serialized. The `test_od_get_rest_api_health_check_response` test confirms that transients return unserialized `WP_Error` instances.